watertap/unit_models/zero_order/tests/test_ion_exchange_zo.py

```diff
@@ -284,3 +307,52 @@ def test_load_parameters(self, model, subtype):
         for (t, j), v in model.fs.unit.removal_frac_mass_solute.items():
             assert v.fixed
             assert v.value == data["removal_frac_mass_solute"][j]["value"]
+
+
+@pytest.mark.parametrize("subtype", [k for k in params.keys()])
+@pytest.mark.component
+def test_costing(subtype):
+    m = ConcreteModel()
+    m.db = Database()
+
+    m.fs = FlowsheetBlock(default={"dynamic": False})
+
+    m.fs.params = WaterParameterBlock(
+        default={"solute_list": ["sulfur", "toc", "tds"]})
+
+    m.fs.costing = ZeroOrderCosting()
+
+    m.fs.unit1 = IonExchangeZO(default={
+        "property_package": m.fs.params,
+        "database": m.db,
+        "process_subtype": subtype})
+
+    m.fs.unit1.inlet.flow_mass_comp[0, "H2O"].fix(10000)
+    m.fs.unit1.inlet.flow_mass_comp[0, "sulfur"].fix(1)
+    m.fs.unit1.inlet.flow_mass_comp[0, "toc"].fix(2)
+    m.fs.unit1.inlet.flow_mass_comp[0, "tds"].fix(3)
+    m.fs.unit1.load_parameters_from_database(use_default_removal=True)
+    assert degrees_of_freedom(m.fs.unit1) == 0
+
+    m.fs.unit1.costing = UnitModelCostingBlock(default={
+        "flowsheet_costing_block": m.fs.costing})
+
+    assert isinstance(m.fs.costing.ion_exchange, Block)
+    assert isinstance(m.fs.costing.ion_exchange.capital_a_parameter,
+                      Var)
+    assert isinstance(m.fs.costing.ion_exchange.capital_b_parameter,
+                      Var)
+    assert isinstance(m.fs.costing.ion_exchange.capital_c_parameter,
+                      Var)
+    assert isinstance(m.fs.costing.ion_exchange.capital_d_parameter,
+                      Var)
+
+    assert isinstance(m.fs.unit1.costing.capital_cost, Var)
+    assert isinstance(m.fs.unit1.costing.capital_cost_constraint,
+                      Constraint)
+
+    assert_units_consistent(m.fs)
+    assert degrees_of_freedom(m.fs.unit1) == 0
+
+    assert m.fs.unit1.electricity[0] in \
```

Comment:
Check that `sodium_chloride` and `ion_exchange_resin` are in registered flows as well?